spec/graphql/mutations/add_barn_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+module Mutations
+  RSpec.describe AddBarn, type: :request do
+    describe '.resolve' do
+      it 'can create a barn' do
+        barn = create(:barn)
+
+        post '/graphql', params: { params: { name: barn.name }}
+
+        expect(barn.reload).to have_attributes(
+          'id'              => barn.id,
+          'name'            => barn.name
+        )
+      end
```

Comment:
Since you are creating the barn at the top of this spec, it's not testing that it's created. Typically in request specs I don't see direct manipulation of models.

Could you do something like this?

```rb
name = "New Barn"
post '/graphql', params: { params: { name: name }}
barn = Barn.find_by name: name
expect(response).to eq(id: barn.id, name: name)
```